src/queries/release.tsx

```diff
@@ -46,7 +46,7 @@ function useReleasesQuery(
 				releases.push(...response.data.filter(isStableRelease))
 
 				const isMaxAutoPaginationReached =
-					paginationCount === MAX_AUTO_PAGINATION
+					paginationCount >= MAX_AUTO_PAGINATION
```

Comment:
This stupid mistake was making the comparator to request way more pages than necessary when needing more than first 1000 results.

The ability to keep paginating was introduced in #741. However, it was suspiciously slow for the given example `/comparator?repo=renovatebot%2Frenovate&to=24.22.3&from=23.95.0`

If you try that preloaded URL in the review environment for this PR, you'll notice the changelog comparison is loaded much faster than before (e.g. production right now).

@HonkingGoose just FYI: when this gets merged, renovate comparison should be much faster.

Reply:
Good catch! 🥳